Cron/CancelPendingOrders.php

```diff
@@ -101,24 +107,50 @@ public function execute()
             $this->logger->info("Cronjob: Cancel Pending Order Cron started.");
             $dateTimeCheck = date('Y-m-d H:i:s', strtotime('-' . $this->pendingOrderTimeout . ' minutes'));
             $sortOrder = $this->sortOrderBuilder->setField('entity_id')->setDirection('DESC')->create();
-            $searchCriteria = $this->searchCriteriaBuilder
-            ->addFilter(
-                'updated_at',
-                $dateTimeCheck,
-                'lt'
-            )->addFilter(
-               'status',
-               static::STATUS_PENDING,
-               'eq'
-            )->setSortOrders(
-                [$sortOrder]
-            )->create();
+
+            if ($this->isCancelPendingOrderAgeEnabled === true
+                && $this->pendingOrderAge > $this->pendingOrderTimeout)
+            {
+                $this->logger->info("Cronjob: PendingOrderAge Enabled.");
+                $pendingOrderAgeCheck = date('Y-m-d H:i:s', strtotime('-' . $this->pendingOrderAge . ' minutes'));
+                $searchCriteria = $this->searchCriteriaBuilder
```

Comment:
@rohitcbr can we move this querying to a separate function.

Reply:
@yashgit891 added a function called getSearchCriteria, used it for pendingOrderCron and resetOrderCron